Files/Views/LayoutModes/GridViewBrowser.xaml.cs

```diff
@@ -574,5 +538,47 @@ private void Grid_Loaded(object sender, RoutedEventArgs e)
             var itemContainer = item as GridViewItem;
             itemContainer.ContextFlyout = ItemContextMenuFlyout;
         }
+
+        private void FileList_ContainerContentChanging(ListViewBase sender, ContainerContentChangingEventArgs args)
+        {
+            if (!args.InRecycleQueue)
+            {
+                // This resizes the items after the item template has been changed and reloaded
+                if (itemTemplateChanging)
+                {
+                    itemTemplateChanging = false;
+                    Behaviors.StretchedGridViewItems.ResizeItems(FileList);
+                }
+
+                InitializeDrag(args.ItemContainer);
+                if (args.Item is ListedItem item && !item.ItemPropertiesInitialized)
+                {
+                    args.ItemContainer.PointerPressed += FileListGridItem_PointerPressed;
+                    args.ItemContainer.CanDrag = args.ItemContainer.IsSelected; // Update CanDrag
+
+
+                    args.RegisterUpdateCallback(3, async (s, c) => {
+                        item.ItemPropertiesInitialized = true;
+                        await ParentShellPageInstance.FilesystemViewModel.LoadExtendedItemProperties(item, currentIconSize);
+                    });
+                }
+                else if (args.Item is ListedItem item1 && item1.ItemPropertiesInitialized && item1.FileImage == null)
+                {
+                    args.RegisterUpdateCallback(3, async (s, c) => {
+                        await ParentShellPageInstance.FilesystemViewModel.LoadItemThumbnail(item1, currentIconSize);
+                    });
+                }
+            }
+            else
+            {
+                UninitializeDrag(args.ItemContainer);
+                args.ItemContainer.PointerPressed -= FileListGridItem_PointerPressed;
+
+                if (args.Item is ListedItem item && item.ItemPropertiesInitialized && item.FileImage != null)
+                {
+                    args.RegisterUpdateCallback(4, (s, c) => ParentShellPageInstance.FilesystemViewModel.UnloadItemThumbnail(item));
```

Comment:
The register update callback is important because it leverages the UI phases we've already established.

Reply:
It looks like `UnloadItemThumbnail` is never called if you try to call it using RegisterUpdateCallback.